fastly/c-at-e/src/index.js

```diff
@@ -1,115 +1,87 @@
+app.get("/https://cdn.polyfill.io", c => c.notFound());
+app.get("/https://polyfill.io", c => c.notFound());
+app.get("/pages/fixedData", c => c.notFound());
+app.get("/v1", (c) => c.redirect("/v3/", 301));
+app.get("/v1/*", (c) => {
+  const requestURL = new URL(c.req.url);
+  requestURL.searchParams.delete("libVersion");
+  requestURL.searchParams.delete("gated");
   //   res.body = "API version 1 has been decommissioned. Your request is being redirected to v2.  The `libVersion` and `gated` query string parameters are no longer supported and if present have been removed from your request.\n\nA deprecation period for v1 existed between August and December 2015, during which time v1 requests were honoured but a deprecation warning was added to output.";
-  return response.redirect(
-    request.url.pathname.replace("/v1", "/v2") + request.url.search
+  return c.redirect(
+    requestURL.pathname.replace("/v1", "/v2") + requestURL.search, 301
   );
 });
 
-router.route("OPTIONS", "*", async (_, response) => {
-  response.setHeader("allow", "OPTIONS, GET, HEAD");
-  response.send("");
+app.options("*", (c) => {
+  c.res.headers.set("allow", "OPTIONS, GET, HEAD");
+  return c.text("");
 });
 
-function configureV2Defaults(request) {
-  // # Override the v3 defaults with the defaults of v2
-  request.url.pathname = request.url.pathname.replace("/v2", "/v3");
-  request.url.searchParams.set("version", "3.25.1");
-  if (!request.url.searchParams.has("unknown")) {
-    request.url.searchParams.set("unknown", "ignore");
+function configureV2Defaults(url) {
+  url.pathname = url.pathname.replace("/v2", "/v3");
+  url.searchParams.set("version", "3.25.1");
+  if (!url.searchParams.has("unknown")) {
+    url.searchParams.set("unknown", "ignore");
   }
+  return url;
 }
 
-router.get(
+app.get(
   "/v3/normalise_querystring_parameters_for_polyfill_bundle",
-  async function (request, response) {
-    response.json(
+  function (c) {
+    return c.json(
       Object.fromEntries(Array.from(
         normalise_querystring_parameters_for_polyfill_bundle(
-          request,
-          request.url.searchParams
+          c.req,
+          new URL(c.req.url).searchParams
         ).entries())
       )
     );
   }
 );
 
-router.route("*", "*", async function (request, response) {
-	console.log(request.url)
-  if (!allowed_methods.has(request.method)) {
-    response.status = 405;
-    return response.send(`${request.method} METHOD NOT ALLOWED`);
-  }
-
-  if (request.method === "PURGE") {
-		request.headers['Fastly-Purge-Requires-Auth'] = "1";
-    let backendResponse = await fetchWithFailover(request.url.toString(), {
-      backend: "polyfill",
-      headers: request.headers,
-      method: request.method
-    });
-    return response.send(backendResponse);
-  }
-
-  const host = request.headers.host;
+app.get("*", handler);
+app.head("*", handler);
+app.options("*", handler);
+app.all("*", c => {
```

Comment:
Looks like this will return 405 for "FASTLYPURGE" & "PURGE" methods now, is that true/intentional?

Reply:
That's true, I can add back support for individual URL purges if we want them. The way I've been purging the site is via the fastly purge API on api.fastly.com and not with the purge http methods

Reply:
Happy to purge through the fastly api :)